### PR TITLE
Get all recommendations API added

### PR DESCRIPTION
**Pull request by Zhaozhi233**
I added an API to get all the recommendations in the databases, when we call, 
'''
/recommendations
'''
There is one question about GET from empty database. 
According to professor's example, it also return a message 200.
So I didn't add 204 for this case.

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -141,6 +141,18 @@ def test_create_recommendation(self):
...
+        """ Get all recommendations tests"""
+        # Test for empty database
+        resp = self.app.get("/recommendations")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+        # Test for non-empty database
+        recommendations = self._create_recommendations(count=5, by_status=True)
+        resp = self.app.get("/recommendations")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(resp.get_json()), 5)
```
Comment (sushanthsamala) [inline tests/test_service.py:154]: Can you also check if the returned recommendations are the same as the created recommendations at line 151?

Comment (Zhaozhi233) [reply tests/test_service.py:154]: No problem!

File: service/service.py [hunk 2]
```diff
@@ -26,6 +26,33 @@ def index():
...
+    
+    With HTTP_200_OK status
+    """
+    app.logger.info("Request for all recommendations in the database")
+
+    recommendations = Recommendation.all()
+    
+    result = []
+    for rec in recommendations:
+        record = {"product_id": rec.id, "related_product_id": rec.rel_id, "type": rec.typeid, "status": rec.status}
```
Comment (sushanthsamala) [inline service/service.py]: You can use 'record = rec.serialize()' instead

File: tests/test_service.py (hunk 1 above)
Comment (sushanthsamala) [reply tests/test_service.py:154]: Greate work George

File: tests/test_service.py [hunk 3]
```diff
@@ -141,6 +141,35 @@ def test_create_recommendation(self):
...
+
+        # Test for the accuracy
+        for i in range(5):
+            product_id = recommendations[i][0].id
```
Comment (sushanthsamala) [inline tests/test_service.py]: You can do something like this 
```
resp = resp.get_json()
for i in range(5):
    returned_recommendation = Recommedation()
    returned_recommendation.deserialize(resp[i])
    self.assertEqual(recommendations[i][0], returned_recommendation)
```

**Review by sushanthsamala — APPROVED** [review]

**Pull request merged**